src/sentry/services/hybrid_cloud/organization/impl.py

```diff
@@ -600,6 +611,45 @@ def upsert_replicated_auth_identity(
 
             existing.update(**update)
 
+    def delete_organization(
+        self, *, organization_id: int, user: RpcUser
+    ) -> RpcOrganizationDeleteResponse:
+        orm_organization = Organization.objects.get(id=organization_id)
+        if orm_organization.is_default:
+            return RpcOrganizationDeleteResponse(
+                response_state=RpcOrganizationDeleteState.CANNOT_REMOVE_DEFAULT_ORG
+            )
+
+        published_sentry_apps = app_service.get_published_sentry_apps_for_organization(
+            organization_id=orm_organization.id
+        )
+
+        if len(published_sentry_apps) > 0:
+            return RpcOrganizationDeleteResponse(
+                response_state=RpcOrganizationDeleteState.OWNS_PUBLISHED_INTEGRATION
+            )
+
+        with transaction.atomic(router.db_for_write(RegionScheduledDeletion)):
+            updated_organization = mark_organization_as_pending_deletion_with_outbox_message(
+                org_id=orm_organization.id
+            )
+
+            if updated_organization is not None:
+                schedule = RegionScheduledDeletion.schedule(orm_organization, days=1, actor=user)
+
+                Organization.objects.uncache_object(updated_organization.id)
```

Comment:
Longer term, it would be nice to have this service method take on creating audit logs and sending emails. Having all of the domain operations together would be nice.

Reply:
👍